app/actions/TrezorActions.js

```diff
@@ -892,3 +911,299 @@ export const getWalletCreationMasterPubKey = () => async (
     throw error;
   };
 };
+
+export const TRZ_PURCHASETICKET_ATTEMPT = "TRZ_PURCHASETICKET_ATTEMPT";
+export const TRZ_PURCHASETICKET_FAILED = "TRZ_PURCHASETICKET_FAILED";
+export const TRZ_PURCHASETICKET_SUCCESS = "TRZ_PURCHASETICKET_SUCCESS";
+
+export const purchaseTicketsV3 = (
+    accountNum,
+    numTickets,
+    vsp
+  ) => async (dispatch, getState) => {
+  dispatch({ type: TRZ_PURCHASETICKET_ATTEMPT });
+
+  if (noDevice(getState)) {
+    dispatch({
+      error: "Device not connected",
+      type: TRZ_PURCHASETICKET_FAILED
+    });
+    return;
+  }
+
+  const {
+    grpc: { walletService }
+  } = getState();
+  const chainParams = selectors.chainParams(getState());
+
+  try {
+    // TODO: Enable on mainnet. The following todo on crypto magic must be
+    // implemented first. Revocation logic and a re-fee payment method must be
+    // added.
+    if (chainParams.trezorCoinName != "Decred Testnet") throw "can only be used on testnet";
+    // TODO: Fill this with deterministic crypto magic.
+    const privateKeySeed = Buffer.alloc(32);
+    privateKeySeed[31] = 1;
+    const net = conToTrezCoinParams(chainParams.trezorCoinName);
+    const ecp = ECPair.fromPrivateKeyBuffer(privateKeySeed, net);
+    const votingKey = ecp.toWIF();
+    const votingAddr = ecp.getAddress();
+    // TODO: Add cspp purchasing.
+    const res = await wallet.purchaseTicketsV3(walletService, "", accountNum, numTickets, false, vsp, {});
+    const splitTx = res.getSplitTx();
+    const decodedInp = await wallet.decodeTransactionLocal(
+      splitTx,
+      chainParams
+    );
+    let changeIndex = 0;
+    for (const out of decodedInp.outputs) {
+      const addr = out.decodedScript.address;
+      const addrValidResp = await wallet.validateAddress(walletService, addr);
+      if (addrValidResp.getIsInternal()) {
+        changeIndex = addrValidResp.getIndex();
+        break;
+      }
+    };
+    await signTransactionAttemptTrezor(splitTx, changeIndex)(dispatch, getState);
+    const refTxs = await walletTxToRefTx(walletService, decodedInp);
+
+    for (const ticket of res.getTicketsList()) {
+      const decodedTicket = await wallet.decodeTransactionLocal(
+        ticket,
+        chainParams
+      );
+      refTxs.hash = decodedTicket.inputs[0].prevTxId;
```

Comment:
Hash doesn't come back with decodeTransactionLocal? Should I be using decodeTransaction? (I think that exists)

Reply:
Making the hash locally.